### PR TITLE
Bump pylint, astroid, fix new detections

### PR DESCRIPTION
**Pull request by VladimirSlavik**
This resolves the pylint and astroid PRs as well as new detections. Not marking as infra since there are code changes too.

Obsoletes: #4541, #4542

### CODE REVIEW

File: pyanaconda/modules/timezone/timezone.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def process_kickstart(self, data):
                 source.type = TIME_SOURCE_POOL
                 source.hostname = source_data.ntp_pool
             else:
-                KickstartParseError(
+                raise KickstartParseError(
```
Comment (poncovka) [inline pyanaconda/modules/timezone/timezone.py:104]: I don't know if we shouldn't delete this part. We are not able to test it.

**Review by poncovka — COMMENTED** [review]
Thanks for working on this.

File: pyanaconda/modules/timezone/timezone.py (hunk 1 above)
Comment (VladimirSlavik) [reply pyanaconda/modules/timezone/timezone.py:104]: I didn't look that far, just wanted to make tests pass...

**Comment by VladimirSlavik** [comment]
Rebased to see if that fixes the mysteriously failing cockpit tests (should not be affected).

**Comment by VladimirSlavik** [comment]
Looks like the answer is "new font makes screenshots go boom" and is completely unrelated.

**Comment by VladimirSlavik** [comment]
Supersedes also #4552 & #4553.

**Review by M4rtinK — APPROVED** [review]
Looks good to me! :)

**Review by poncovka — APPROVED** [review]

**Pull request merged**